capstone/src/main/webapp/loginAuth.js

```diff
@@ -1,12 +1,23 @@
 function fetchLoginStatus() {
-  fetch('/login').then(response => response.text()).then((loginStatus) => {
-    //Because loginStatus will return either the logout / login message
-    //we can just check the length of return message to ascertain if
-    //we are logged in or out
-    if(loginStatus.length < 100) {
-      window.location = "/login.html";
+  fetch("/login").then(response => response.json()).then((loginStatus) => {
+    const redirectUrl = document.createElement('a');
+    redirectUrl.setAttribute('href', loginStatus.url);
```

Comment:
Thank you for making this change! This makes it easier for us to see what is going on in your logic here. I like this method a lot!